self_hosted/main.jou

```diff
@@ -467,6 +483,13 @@ def main(argc: int, argv: byte**) -> int:
         compiler.process_imports_and_exports()
         compiler.typecheck_stage3_all_files()
 
+        mainfile = find_file(compiler.files, compiler.nfiles, args.main_path)
+        assert mainfile
```

Comment:
Jou is a more explicit language than C when it comes to checking for NULL. In C, `if(some_pointer)` is same as `if(some_pointer != NULL)`. In Jou you need to compare to NULL directly:

```suggestion
        assert mainfile != NULL
```

Reply:
I forgot it add it after test